src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/Tensor.cs

```diff
@@ -691,6 +691,55 @@ private Tensor<T> GetTriangle(int offset, bool upper)
         /// <param name="value">The new value to set at the specified position in this Tensor.</param>
         public abstract void SetValue(int index, T value);
 
+        /// <summary>
+        /// The type that implements enumerators for <see cref="Tensor{T}"/> instances.
+        /// </summary>
+        public struct Enumerator : IEnumerator<T>
+        {
+            private readonly Tensor<T> tensor;
+            private int index;
+
+            internal Enumerator(Tensor<T> tensor)
+            {
+                Debug.Assert(tensor != null);
+
+                this.tensor = tensor;
+                index = 0;
+                Current = default;
+            }
+
+            public T Current { get; private set; }
+
+            public bool MoveNext()
+            {
+                if (index < tensor.Length)
+                {
+                    Current = tensor.GetValue(index);
+                    ++index;
+                    return true;
+                }
+                else
+                {
+                    Current = default;
+                    return false;
+                }
+            }
+
+            object? IEnumerator.Current => Current;
```

Comment:
This shouldn't be needed.  If removing it results in errors, I expect the problem is the use of the auto-prop on Current including a private set; it'd be better to instead just write out the field manually and remove the setter from the property:
```C#
private T _current;
...
public T Current => _current;
```

Reply:
Hmm it still gives me an error when I remove the explicit implementation of `Current`, even when I remove the auto-prop and create the backing field explicitly. 
`Error: CS0738	'Tensor<T>.Enumerator' does not implement interface member 'IEnumerator.Current'. 'Tensor<T>.Enumerator.Current' cannot implement 'IEnumerator.Current' because it does not have the matching return type of 'object'.	System.Numerics.Tensors (net461), System.Numerics.Tensors (netstandard1.1), System.Numerics.Tensors (netstandard2.0)`

Reply:
Ah, right.  Ignore me: hadn't had my coffee yet :)